blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractPipelineImpl.java

```diff
@@ -231,6 +233,22 @@ public Link getLink() {
         };
     }
 
+    @Override
+    public List<Object> getParameters() {
+        return getParameterDefinitions(job);
+    }
+
+    public static List<Object> getParameterDefinitions(Job job){
```

Comment:
@vivek @kzantow the declarative model can give us the parameter definitions without having to do an initial run , as described in [JENKINS-40241](https://issues.jenkins-ci.org/browse/JENKINS-40241). Do you think this is worth exploring? What are the advantages of reading from PMD for this info if it is available? Can you see any roadblocks for making this work?